File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/FAQResponse.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import java.util.UUID;

import dev.jefy.connectpro.portfolio.domain.model.FAQ;

/**
 * @author Jôph Yamba
 */
public record FAQResponse (
    UUID id,
    String question,
    String answer
) {
    public static FAQResponse from(FAQ faq) {
        return new FAQResponse(
                faq.getId().value(),
                faq.getQuestion(),
                faq.getAnswer()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/FAQResponse.kt
package dev.jefy.connectpro.portfolio.application.dtos

import dev.jefy.connectpro.portfolio.domain.model.FAQ
import java.util.*


/**
 * @author Jôph Yamba
 */
data class FAQResponse(val id: UUID, val question: String, val answer: String) 

fun FAQ.toResponse(): FAQResponse =FAQResponse(
    id = this.id.value,
    question = this.question,
    answer = this.answer
)

